builders/pallets-precompiles/precompiles/relay-data-verifier.md

```diff
@@ -0,0 +1,149 @@
+
+???+ function "**verifyEntry**(_uint32_ relayBlockNumber, _ReadProof_ calldata readProof, _bytes_ callData key) — Verifies a storage entry in the relay chain using a relay block number, a storage proof, and the storage key. It returns the value associated with the key if the verification is successful"
+
+    === "Parameters"
+
+          - `relayBlockNumber` - the relay block number for which the data is being verified. The latest relay block number can be obtained from the `latestRelayBlockNumber()` function
+
+          - `readProof` - a struct defined in the precompile contract, containing the storage proof used to verify the data. The `ReadProof` struct is defined as:
+          ```
+          struct ReadProof {
+                    // The block hash against which the proof is generated
+                    bytes32 at;
+                    /// The storage proof
+                    bytes[] proof;
+               }
+          ```
+
+          - `key` - the storage key for the generated proof
+
+???+ function "**verifyEntries**(_uint32_ relayBlockNumber, _ReadProof_ calldata readProof, _bytes[]_ callData keys) — Verifies a set of entries in the relay chain and returns the corresponding values. This function takes a relay block number, a storage proof, and an array of storage keys to verify. It returns an array of values associated with the keys, in the same order as the keys"
+
+    === "Parameters"
+
+          - `relayBlockNumber` - The relay block number for which the data is being verified. The latest relay block number can be obtained from the `latestRelayBlockNumber()` function
+
+          - `readProof` - A struct defined in the precompile contract, containing the storage proof used to verify the data. The `ReadProof` struct is defined as:
+          ```
+          struct ReadProof {
+                    // The block hash against which the proof is generated
+                    bytes32 at;
+                    /// The storage proof
+                    bytes[] proof;
+               }
+          ```
+
+          - `keys` - The storage keys for the generated proof
+
+## Interact with the Solidity Interface {: #interact-with-the-solidity-interface }
+
+A typical workflow to verify relay chain data involves the following steps:
+
+1. **Moonbeam RPC Call** - Call the `latestRelayBlockNumber` function to get the latest relay block number tracked by the chain in the `pallet-storage-root`
+
+2. **Relay RPC Call** - Call the `chain_getBlockHash(blockNumber)` RPC method to get the relay block hash for the block number obtained in step 1
+
+3. **Relay RPC Call** - Call the `state_getReadProof(keys, at)` RPC method to retrieve the storage proof, where `at` is the relay block hash obtained in step 2, `keys` is an Array of strings which contains the keys for target storage items. For `@polkadot/api`, it can be obtained via `api.query.module.key()` function
```

Comment:
```suggestion
3. **Relay RPC Call** - call the `state_getReadProof(keys, at)` RPC method to retrieve the storage proof, where `at` is the relay block hash obtained in step 2, `keys` is an Array of strings which contains the keys for target storage items. For `@polkadot/api`, it can be obtained via `api.query.module.key()` function
```